### PR TITLE
test: Aura min improve l2 handling

### PR DESCRIPTION
**Pull request by Tritium-VLK**
Do 2 passes of Aura redistribution.  The first with a buffer to allow poosl close to the line to pass over it.  Then a second with no buffer to ensure the desired end state is reached with the widest possible inclusion rate. 

Also improve dependancies

### CODE REVIEW

**Comment by Tritium-VLK** [comment]
Run results:
https://github.com/BalancerMaxis/protocol_fee_allocator/blob/d18629ed2f86c6736871ec8cb1879daf28ba2cb8/fee_allocator/allocations/incentives_2024-04-25_2024-05-09.csv

File: fee_allocator/accounting/distribution.py [hunk 1]
```diff
@@ -117,60 +175,13 @@ def re_distribute_incentives(
...
-            if debt_to_aura_market - debt_repaid >= 0:
-                print(
-                    f"{incentives[pool_id]['chain']}:{pool_id}  remaining debt to aura market: {debt_to_aura_market}, Debt repaid: {debt_repaid}, debt remaining: {debt_to_aura_market - debt_repaid}"
-                )
-    return incentives
+    # Now after everything is done, we need to make sure that all pools have at least min_aura_incentive
+    # if not we need to redistribute all aura_incentives to bal_incentives for that pool and keep track of how much has been reallocated
+
+    # First redistribute with a 30% buffer.
+    result = handle_aura_min(incentives, min_aura_incentive * Decimal(0.7))
```
Comment (Tritium-VLK) [inline fee_allocator/accounting/distribution.py:182]:
```suggestion
    result = handle_aura_min(incentives, min_aura_incentive * Decimal(0.5))
```

Try 50% buffer

**Comment by Tritium-VLK** [comment]
25% seems like a good starting buffer for now.  Can think about making that more dynamic or adjusting it later.   In the end this PR results in Aura bribs going to 3 pools on arbiturm instead of 1, and basically allows pools that are near the split to gain a little from pools that are not near and make it over before surrendering their aura incentives to larger pools.

File: main.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def main() -> None:
     )
     output_file_name = parser.parse_args().output_file_name or "current_fees.csv"
     fees_file_name = parser.parse_args().fees_file_name or "current_fees_collected.json"
-    fees_path = os.path.join(ROOT, "fee_allocator", "fees_collected", fees_file_name)
+    fees_path = f"fee_allocator/fees_collected/{fees_file_name}"
```
Comment (gosuto-inzasheru) [inline main.py:88]: why this change? `os.path.join` works better in some cases (mostly different file systems)

File: requirements.txt [hunk 3]
```diff
@@ -4,4 +4,5 @@ gql[requests]==3.4.1
 pycoingecko==3.1.0
 pandas==2.0.3
 simplejson==3.19.2
-git+https://github.com/BalancerMaxis/bal_addresses@0.9.1
+git+https://github.com/BalancerMaxis/bal_addresses@0.9.4
+git+https://github.com/BalancerMaxis/bal_tools@v0.0.1
```
Comment (gosuto-inzasheru) [inline requirements.txt:8]: bal tools is at v0.0.3 already.

bump it, or maybe just unpin it?

File: fee_allocator/accounting/distribution.py [hunk 4]
```diff
@@ -117,60 +175,18 @@ def re_distribute_incentives(
...
-                )
-            if debt_to_aura_market - debt_repaid >= 0:
-                print(
-                    f"{incentives[pool_id]['chain']}:{pool_id}  remaining debt to aura market: {debt_to_aura_market}, Debt repaid: {debt_repaid}, debt remaining: {debt_to_aura_market - debt_repaid}"
-                )
-    return incentives
+    # Now after everything is done, we need to make sure that all pools have at least min_aura_incentive
+    # if not we need to redistribute all aura_incentives to bal_incentives for that pool and keep track of how much has been reallocated
+
+    first_pass_buffer = 0.25
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/distribution.py:181]: this could be a param of the function maybe

File: fee_allocator/accounting/distribution.py [hunk 5]
```diff
@@ -117,60 +175,18 @@ def re_distribute_incentives(
...
+    # First redistribute with a % buffer.
+    print(f"Redistributing Aura with a {first_pass_buffer} buffer.")
+    result = handle_aura_min(
+        incentives, min_aura_incentive * Decimal(1 - first_pass_buffer)
+    )
+    # Now redistribute again with no buffer
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/distribution.py:187]: nice solution!

i guess an even more inclusive approach would be to do this incrementally, so do a pass at 10%, 20%, 30% ... all the way up to 100%?

**Review by gosuto-inzasheru — COMMENTED** [review]
minor notes

File: main.py (hunk 2 above)
Comment (Tritium-VLK) [reply main.py:88]: It wasn't working.  I changed it to make it find the file.   Somehow the python version upgrade broke it.  You're welcome to try if you can figure it out.

File: requirements.txt (hunk 3 above)
Comment (Tritium-VLK) [reply requirements.txt:8]: Because we are not using pypi bal_tools either has to not be provided, or the @ string has to match the current version in that version of bal_addresses. bal_addresses@0.9.1 still uses ball tools 0.0.1

File: fee_allocator/accounting/distribution.py (hunk 5 above)
Comment (Tritium-VLK) [reply fee_allocator/accounting/distribution.py:187]: Yeah we could, and maybe shoulkd later.  I don't think it will make any diference right now tho, In general there's not that many pools left.

File: requirements.txt (hunk 3 above)
Comment (gosuto-inzasheru) [reply requirements.txt:8]: but you are bumping bal addresses to 0.9.3 here, which uses bal_tools@v0.0.3

Comment (gosuto-inzasheru) [reply requirements.txt:8]: ah no wait that is not true; main branch is using v0.0.3 but v0.9.4 is still using v0.0.1

File: main.py (hunk 2 above)
Comment (gosuto-inzasheru) [reply main.py:88]: ah ok well in that case changing it is a good reason :)

**Comment by Tritium-VLK** [comment]
@gosuto-inzasheru want to do one final test run before merge, but lets make sure all review commetns are done first and reviews have passed and then I'll kick off a final run before merge.

File: fee_allocator/accounting/distribution.py (hunk 5 above)
Comment (gosuto-inzasheru) [reply fee_allocator/accounting/distribution.py:186]: yes, agree that this is a nice to have that can always be added later when needed

**Review by gosuto-inzasheru — APPROVED** [review]
lgtm! one more test run is a good idea
